src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -195,12 +239,36 @@ def _parse_frontend_config(config):
              whether SSL is used.
     :rtype: (int, boolean)
     """
-    match = re.search(r'port=(.*:)?(\d+)(s)?', config)
+    match = re.search(r'^(beast|civetweb)\s+.+$', config)
     if match:
-        port = int(match.group(2))
-        ssl = match.group(3) == 's'
-        return port, ssl
-    raise LookupError('Failed to determine RGW port')
+        if match.group(1) == 'beast':
+            match = re.search(r'(port|ssl_port|endpoint|ssl_endpoint)=(.+)',
+                              config)
+            if match:
+                option_name = match.group(1)
+                if option_name in ['port', 'ssl_port']:
+                    match = re.search(r'(\d+)', match.group(2))
+                    if match:
+                        port = int(match.group(1))
+                        ssl = option_name == 'ssl_port'
+                        return port, ssl
+                if option_name in ['endpoint', 'ssl_endpoint']:
+                    match = re.search(r'([\d.]+|\[.+\])(:(\d+))?',
+                                      match.group(2))
+                    if match:
+                        port = int(match.group(3)) if \
+                            match.group(2) is not None else 443 if \
+                            option_name == 'ssl_endpoint' else \
+                            80
+                        ssl = option_name == 'ssl_endpoint'
+                        return port, ssl
```

Comment:
Have you considered using standard module [`configparser`](https://docs.python.org/3/library/configparser.html)?

It's basically an INI parser, you just need to prepend a section (`[whatever]`, as it doesn't support section-less INI files). That will save you the hassle (and the risks) of dealing with custom parsing.
```py
def _parse_frontend_config(config):
  from configparser import ConfigParser

  cp = ConfigParser(
    strict=False,    # To allow option duplicates, as in `port=80 port=8080`, otherwise it'll raise an exception
    allow_no_value=True,    # To allow options with no value set
    delimiters=('=',),    # Only '=' is accepted to separate key-values
    defaults={'ssl_port': 443}    # defaults in case option is not specified
  )
  SECTION = 'rgw_client_config'
  cp.read_string(f"[{SECTION}] {config}")

  if cp.has_option(SECTION, 'beast'):
    if cp.has_option(SECTION, 'ssl_port'):
      port = cp.getint(SECTION, 'ssl_port', 
    ...
  elif cp.has_option(SECTION, 'civetweb'):
    ...
```

Reply:
No, but IMO looking into your `configparsers` example i don't think it will make things easier. Finally you need to use regex to parse the value.